smart_card_connector_app/src/window-apps-displaying.js

```diff
@@ -119,12 +119,37 @@ function initializeWithBackgroundPage(backgroundPage) {
   goog.asserts.assert(backgroundPage);
 
   goog.log.fine(logger, 'Registering listener on connected clients update');
-  // FIXME(emaxx): Do unsubscription too.
-  const subscriber =
+
+  /**
+   * Points to the "addOnUpdateListener" method of the AppList instance
+   * that is owned by the background page.
+   */
+  const appListSubscriber =
       /** @type {function(function(!Array.<string>))} */
       (GSC.ObjectHelpers.extractKey(
           backgroundPage, 'googleSmartCard_clientAppListUpdateSubscriber'));
-  subscriber(onUpdateListener);
+  // Start tracking the current list of apps.
+  appListSubscriber(onUpdateListener);
+
+  /**
+   * Points to the "removeOnUpdateListener" method of the AppList instance
+   * that is owned by the background page.
+   */
+  const appListUnsubscriber =
+      /** @type {function(function(!Array.<string>))} */
+      (GSC.ObjectHelpers.extractKey(
```

Comment:
maybe it was indirectly called by some other file? because it was also used for subscribing (which already existed in the code) and was released before

Reply:
I think indirect inclusions don't count; I remember you even reported some time ago a runtime error, and it turned out to be caused by a missing goog.require clause. So let's add all needed ones explicitly.

Reply:
could errors with pthreads somehow be caused because of a goog.require? because I'm facing such error now even after adding the clause...checking for others but just wondering

Reply:
Hmm, pthreads - probably not... Can you share the error stack trace with me?

Reply:
`Pthread 0x0 completed its main entry point with an `unwind`, keeping the worker alive for asynchronous operation.
threadPrintErr @ executable_module.worker.js:60
self.onmessage @ executable_module.worker.js:191
`

Reply:
Ah I remember seeing these. Not sure if they actually designate any error - at least I haven't observed anything bad happening afterwards. And what I saw was a thread that we wanted to terminate anyway was reporting this, so while it'll be good to investigate it one day it seems likely innocent.